src/Toolkit/Toolkit.Maui/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -176,4 +178,28 @@ private async void Refresh()
     /// </summary>
     [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
     public event System.EventHandler? SourceUpdated;
+
+    private void SymbolDisplay_PropertyChanged(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == nameof(Window) && Window != null)
+        {
+            Refresh();
+            Window.DisplayDensityChanged += Window_DisplayDensityChanged;
```

Comment:
This needs to be a weak event, or this control will stay resident until the window goes out of scope (which usually is never).
Also if window is changing, you'd want to unhook the old handler - that's easier to do with the weak handler which has a reference to the old one.